File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range(len(nums)):
            another = target - nums[i]
            if another in nums:
                try:
                    j = nums.index(another)
                    if i != j:
                        return [i, j]
                except ValueError as e:
                    continue
        return []

File: 0001-two-sum/0001-two-sum.java
import java.util.*;

public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int another = target - nums[i];
            if (numMap.containsKey(another)) {
                return new int[] {numMap.get(another), i};
            }
            numMap.put(nums[i], i);
        } 
        return new int[0];
    }
    
}
